lib/matplotlib/legend.py

```diff
@@ -404,9 +404,9 @@ def __init__(self, parent, handles, labels,
         _lab, _hand = [], []
         for label, handle in zip(labels, handles):
             if isinstance(label, str) and label.startswith('_'):
-                cbook._warn_external('The handle {!r} has a label of {!r} '
-                                     'which cannot be automatically added to'
-                                     ' the legend.'.format(handle, label))
+                cbook._warn_external('The label {!r} of handle {!r} cannot be '
+                                     'a string starting with '
+                                     '"_"'.format(label, handle))
```

Comment:
The merge conflict comes from internal changes `cbook._warn_external` -> `_api.warn_external`.
While you're at it, using f-strings makes this more readable.

```suggestion
                _api.warn_external(f'The label {label!r} of handle {handle!r} '
                                   'cannot be a string starting with "_"')